### ISSUE TITLE
Multiple DataFrames in one sheet

### ISSUE THREAD

**Issue by vishalhedgevantage** [felipenoris/XLSX.jl#180]
I am not able to find a way to write multiple dataframes in one excel sheet. 
Is there any existing method that I can use for doing this?

**Comment by nathanrboyer** [comment]
Yes, there is an `anchor_cell` keyword argument for the `writetable!` function that you can use to write multiple dataframes per sheet.
```julia
using DataFrames, XLSX
df1 = DataFrame(a = [1,2,3], b = [4,5,6])
df2 = DataFrame(c = [7,8,9], d = [10,11,12])
XLSX.openxlsx("path/to/file.xlsx", mode="w") do file
    XLSX.writetable!(file[1], df1)
    XLSX.writetable!(file[1], df2, anchor_cell=XLSX.CellRef("C1")
end
```

**Comment by nilshg** [comment]
I think this can be closed @felipenoris?

**Comment by stevenpkent** [comment]
@nathanrboyer -

I believe I'm encountering a similar issue. I am iterating through a multi-result dataset from a SQL query and I am using XLSX to try and accomplish:
(1) create a new Excel file - done
(2) do the following for each item in the dataset:
    a. create 2 dataframes that are created by 2 more SQL queries - done
    b. add a new worksheet in the Excel workbook - done
    c. save the 2 dataframes in the newly added worksheet
        -the first dataframe needs to be inserted at anchor cell A1
        -the second dataframe needs to be inserted at anchor cell H1

It is item 2c that I have been unable to accomplish. I've had some amount of success but have not figured out how to save both dataframes in every worksheet of the workbook.

Any assistance is appreciated. Also thank you all for creating and maintaining XLSX.

**Comment by stevenpkent** [comment]
> Yes, there is an `anchor_cell` keyword argument for the `writetable!` function that you can use to write multiple dataframes per sheet.
> 
> using DataFrames, XLSX
> df1 = DataFrame(a = [1,2,3], b = [4,5,6])
> df2 = DataFrame(c = [7,8,9], d = [10,11,12])
> XLSX.openxlsx("path/to/file.xlsx", mode="w") do file
>     XLSX.writetable!(file[1], df1)
>     XLSX.writetable!(file[1], df2, anchor_cell=XLSX.CellRef("C1"))
> end

I ended up getting past this issue using `XLSX.writetable!(worksheet, df, anchor_cell)`. What does the `file[1]` mean?

**Comment by nathanrboyer** [comment]
Same thing. `file[1]` is the first sheet in the file. You can access a worksheet by indexing into a file by name or number: `sheet1 = file[1]` or `sheet1 = file["Sheet1"]`. (Then you can access a cell by indexing a worksheet: `cell1 = sheet1[1,1]` or `cell1 = file[1][1,1]`.)
